core-java-concurrency/src/main/java/com/baeldung/concurrent/prioritytaskexecution/PriorityJobScheduler.java

```diff
@@ -9,21 +9,23 @@
 public class PriorityJobScheduler {
 
     private ExecutorService priorityJobPoolExecutor;
-    private ExecutorService priorityJobScheduler;
-    private PriorityBlockingQueue<Runnable> priorityQueue;
+    private ExecutorService priorityJobScheduler = 
+      Executors.newSingleThreadExecutor();
+    private PriorityBlockingQueue<Job> priorityQueue;
 
     public PriorityJobScheduler(Integer poolSize, Integer queueSize) {
         priorityJobPoolExecutor = Executors.newFixedThreadPool(poolSize);
-        Comparator<? super Job> jobComparator = Comparator.comparing(Job::getJobPriority);
-        priorityQueue = new PriorityBlockingQueue<Runnable>(queueSize, 
-          (Comparator<? super Runnable>) jobComparator);
+        priorityQueue = new PriorityBlockingQueue<Job>(queueSize, 
+          Comparator.comparing(Job::getJobPriority));
 
-        priorityJobScheduler = Executors.newSingleThreadExecutor();
         priorityJobScheduler.execute(()->{
             while (true) {
                 try {
                     priorityJobPoolExecutor.execute(priorityQueue.take());
                 } catch (InterruptedException e) {
+                    if (priorityQueue.size() > 0) {
```

Comment:
Do we need that check here?

Reply:
I would prefer to remove it and just break. But then we go back to square one.